src/adapters/methods/lora.py

```diff
@@ -39,58 +40,121 @@ def __init__(
             self.lora_dropout = lambda x: x
 
         # Actual trainable parameters
-        if self.r > 1 and self.composition_mode == "scale":
-            raise ValueError("Can only use composition_mode='scale' when r == 1.")
-        if self.r > 0:
-            if self.composition_mode == "add":
-                self.lora_A = nn.Parameter(torch.zeros(lora_A_shape))
-            self.lora_B = nn.Parameter(torch.zeros(lora_B_shape))
-            self.scaling = self.lora_alpha / self.r
-
-            if self.use_gating:
-                self.gate = nn.Linear(lora_A_shape[-1], gating_heads)
-
-            if config.init_weights == "lora":
-                # initialize A the same way as the default for nn.Linear and B to zero
-                if self.composition_mode == "add":
-                    nn.init.kaiming_uniform_(self.lora_A, a=math.sqrt(5))
-                nn.init.zeros_(self.lora_B)
-                if self.use_gating:
-                    nn.init.normal_(self.gate.weight, std=0.02)
-            elif config.init_weights == "bert":
-                if self.composition_mode == "add":
-                    nn.init.normal_(self.lora_A, std=0.02)
-                nn.init.normal_(self.lora_B, std=0.02)
-                if self.use_gating:
-                    nn.init.normal_(self.gate.weight, std=0.02)
-            elif config.init_weights == "ia3":
-                if self.composition_mode == "add":
-                    nn.init.ones_(self.lora_A)
-                nn.init.ones_(self.lora_B)
-                if self.use_gating:
-                    nn.init.normal_(self.gate.weight, std=0.02)
-            else:
-                raise ValueError("Unknown init_weights type: {}".format(config.init_weights))
+        self.lora_A = nn.Parameter(torch.zeros(lora_A_shape))
+        self.lora_B = nn.Parameter(torch.zeros(lora_B_shape))
+        self.scaling = self.lora_alpha / self.r
+
+        if config.init_weights == "lora":
+            # initialize A the same way as the default for nn.Linear and B to zero
+            nn.init.kaiming_uniform_(self.lora_A, a=math.sqrt(5))
+            nn.init.zeros_(self.lora_B)
+        elif config.init_weights == "bert":
+            nn.init.normal_(self.lora_A, std=0.02)
+            nn.init.normal_(self.lora_B, std=0.02)
+        elif config.init_weights == "ia3":
```

Comment:
I am not super familiar with the initialization methods, but is it intended that lora still has ia3 initialization and the other way around?

Reply:
yes, to avoid throwing an error. Using these combinations might not make sense though.

Reply:
I think we should include this with a small comment for clarity in the code.